src/Datadog.Trace/Configuration/IntegrationIds.cs

```diff
@@ -0,0 +1,24 @@
+// ReSharper disable InconsistentNaming - Name is used for integration names
+namespace Datadog.Trace.Configuration
+{
+    internal enum IntegrationIds
```

Comment:
I'm a little worried about implementing this optimization if we still might have an issue with mismatching versions of `Datadog.Trace` and `Datadog.Trace.ClrProfiler.Managed`. What would it take to move this logic into `Datadog.Trace.ClrProfiler.Managed.Core` or `Datadog.Trace.ClrProfiler.Managed`?

Reply:
I can't, because some integrations (aspnetcore) are defined in Datadog.Trace. But I can try thinking of a system where Datadog.Trace.ClrProfiler.Managed would define its own id and register them

Reply:
I've spent a bit of time on this and I'm really not sure how to solve the problem :/

Reply:
Ok, I added a mechanism to reference integrations either by id or by name. This way, it'll fallback on the name if the integration is not known by that version of Datadog.Trace.

The drawback is that it will crash if the new version of Datadog.Trace.ClrProfiler.Managed is used with a Datadog.Trace that doesn't have the IntegrationRegistry...

Reply:
FWIW, in the document we compiled regarding the mismatching versions, Datadog.Trace.ClrProfiler.Managed will only load the corresponding or newer Datadog.Trace. So this may not be an issue as long we do not introduce and then remove the IntegrationRegistry before we resolve that versioning issue.